enhancements/sdk-fbc-run-bundle.md

```diff
@@ -0,0 +1,279 @@
+
+[Channel](<https://github.com/operator-framework/operator-sdk/blob/18ca92070fb3dd5e9f7d4581b27cc3e167a082d9/internal/olm/operator/bundle/install.go#L260>)
+
+These three structs are used to construct the declarative config object. This declarative config object is a File-Based Catalog representation of an index with a single bundle present in it. This declarative config object can easily be transformed into `YAML/JSON` format.
+
+It is noteworthy that this enhancement proposal does not change the `run bundle` command from a user’s perspective. The command-line arguments and flags will remain the same. However, the Operator SDK will handle adding a newly-generated bundle to an index differently than it currently does. That is the goal of this enhancement proposal.
+
+#### Command Line Interface
+
+The command line interface will remain exactly the same as the original `run bundle` command.
+
+#### Assumptions
+
+Like the existing run bundle command, we **assume** the following:
+
+ 1. OLM is already installed
+ 1. kube API is available with the permissions to create resources
+ 1. kubelet can see and pull the provided images (i.e. bundle and index images)
+ 1. If an index image is provided, we assume that it has:
+    - bash, so we can change the entrypoint
+    - if it is a sqlite index image, we assume that:
+        - the existing registry DB located at `/database/index.db`
+        - dependencies required for `opm registry add` so we can inject the bundle into the DB
+
+#### What happens when the `run bundle` command is invoked?
+
+1. The labels from the bundle annotations are parsed. The annotations are received from the bundle image that is passed in as a required argument for the run bundle command. Operator SDK will parse information like channel name and CSV name from the labels in the bundle annotations.
+
+2. Additionally, the Operator SDK also parses the label on the index image. The label `operators.operatorframework.io.index.configs.v1=/configs` is used to differentiate FBC images from SQLite images. If this label exists on the index image, we have two paths:
+
+    - Create an File-Based Catalog on the fly if an index image is not being passed as user input, i.e. for the default index image.
+    - Add the bundle contents to the index image if a user provides an index image as input.
+
+3. The Operator SDK, using the data gathered from the above steps as parameters, will invoke Operator Registry APIs that will generate three major components in the form of Operator Registry-defined objects - Package, Channel & Bundle.
+
+4. The Operator SDK will convert these objects into more commonly-consumable formats like JSON/YAML blobs and combine these blobs together to generate a basic File-Based Catalog representation of the bundle image. In other words, the simplest File-Based Catalog consists of one package blob, one channel blob, and one bundle blob. These blobs represent the bundle data.
```

Comment:
It's worth mentioning that these are the "generating FBC on the fly" steps that we've been referring to.